tests/devices/unit_tests/test_xspress3.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+
+import pytest
+from ophyd_async.core import DeviceCollector, set_mock_value
+
+from dodal.devices.xspress3.xspress3 import (
+    AcquireState,
+    DetectorState,
+    Xspress3,
+)
+
+
+@pytest.fixture
+async def mock_xspress3mini(prefix: str = "BLXX-EA-DET-007:") -> Xspress3:
+    async with DeviceCollector(mock=True):
+        mock_xspress3mini = Xspress3(prefix, "Xspress3Mini", 2)
+    assert mock_xspress3mini.channels[1].name == "Xspress3Mini-channels-1"
+    assert mock_xspress3mini.channels[2].name == "Xspress3Mini-channels-2"
+    assert (
+        mock_xspress3mini.get_roi_calc_status[1].name
+        == "Xspress3Mini-get_roi_calc_status-1"
+    )
+    assert (
+        mock_xspress3mini.get_roi_calc_status[2].name
+        == "Xspress3Mini-get_roi_calc_status-2"
+    )
+    assert mock_xspress3mini.roi_mca[1].name == "Xspress3Mini-roi_mca-1"
+    assert mock_xspress3mini.roi_mca[2].name == "Xspress3Mini-roi_mca-2"
+
+    return mock_xspress3mini
+
+
+async def test_stage_success_on_busy_state(mock_xspress3mini: Xspress3):
+    set_mock_value(mock_xspress3mini.detector_state, DetectorState.ACQUIRE)
+    status = await mock_xspress3mini.stage()
+    assert status.done is False
+    set_mock_value(mock_xspress3mini.acquire, AcquireState.DONE)
+    await asyncio.sleep(0.1)
+    assert status.done is True
```

Comment:
Must: This test isn't doing what you think it is. if you remove `set_mock_value(mock_xspress3mini.acquire, AcquireState.DONE)` it still passes fine. I think you need to write some tests that confirm:
* That stage is setting everything you think it is
* That if acquire RBV does not go to DONE it fails
* That `RE(bps.stage(...))` is happy

Reply:
`Stageable` is a protocol, meaning you don't actually `need` to inherit from it, you can also type annotate uses of it and then it will be checked via duck typing. If you inherit from it you're just making it clearer in the code: this must be `Stageable`.

Both bullet points are good, I would quite like all devices to have RE tests, it would have caught several issues we've seen on the beamline. I'd also happily support an issue in bluesky to separate out the `Stageable` protocol. Something like:

```python
class Stageable(Protocol):
    def stage(self): ...

class StageableUnstageable(Protocol):
    def stage(self): ...
    def unstage(self): ...
```